src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalServiceException.cs

```diff
@@ -74,37 +74,35 @@ internal AdalServiceException(string errorCode, Exception innerException)
         /// <param name="errorCode">The protocol error code returned by the service or generated by client. This is the code you can rely on for exception handling.</param>
         /// <param name="message">The error message that explains the reason for the exception.</param>
         /// <param name="serviceErrorCodes">The specific error codes that may be returned by the service.</param>
+        /// <param name="response">The HttpResponse returned by the service.</param>
         /// <param name="innerException">The exception that is the cause of the current exception, or a null reference if no inner exception is specified. It may especially contain the actual error message returned by the service.</param>
-        internal AdalServiceException(string errorCode, string message, string[] serviceErrorCodes, Exception innerException)
-            : base(errorCode, message, (innerException is HttpRequestWrapperException) ? innerException.InnerException : innerException)
+        internal AdalServiceException(string errorCode, string message, string[] serviceErrorCodes, IHttpWebResponse response, Exception innerException)
```

Comment:
>string[] serviceErrorCodes, IHttpWebResponse response, [](start = 72, length = 54)

note: it made me a bit confused reading the code that innerException is moving location, though this also makes sense. What is the consideration for keeping innerException at the end?

Reply:
just for consistency, as it was at the end before, but now we need the IHttpWebResponse. i could move that to the end though. Do you have a preference?

---
In reply to: [267508974](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1530#discussion_r267508974) [](ancestors = 267508974)